app/controllers/scheduler_controller.rb

```diff
@@ -5,7 +5,8 @@ class SchedulerController < ApplicationController
   #
   # The jump date should also be set for JS requests, so that the
   # source for the autocomplete can be set to the right date.
-  before_action :set_search_date, only: [:index], :if => [:semester_request?, :xhr_request?]
+  before_action :set_search_date, only: [:index, :create],
+    :if => [:semester_request?, :xhr_request?]
```

Comment:
Align the elements of a hash literal if they span more than one line.<br>Use the new Ruby 1.9 hash syntax.